### DISCUSSION TITLE
Adding App Menus

### DISCUSSION THREAD

**Discussion by dirkeinecke** [slint-ui/slint#6983] [Ideas] [answered]
When creating a GUI app, it would be helpful if Slint provided a way to add menus to the app. But not workarounds—real menus. Especially on macOS, an app's menu is outside the window, and being able to define this through Slint would be cool.

In the meantime: Can anyone recommend crates for creating menus in a Mac app (Rust + Slint)?

**Comment by ogoffart** [comment, marked as answer]
We're working on this.
Status: https://github.com/slint-ui/slint/issues/38#issuecomment-2488195457
This won't be in the Slint 1.9 release, but maybe in Slint 1.10

As of now, you can already get menu using the private API and the `muda` crate.
An example in our repo there: https://github.com/slint-ui/slint/blob/901afde11fda5ad66eb6a2535951034dfbe76462/tools/lsp/preview/native.rs#L269
This was also discussed there: https://github.com/slint-ui/slint/discussions/3640
